### PR TITLE
Chuzzle nested pattern variables when emitting the final column

### PR DESCRIPTION
**Pull request by CodaFi**
### What's in this pull request?

Missed this while implementing RValues the first time around.

### CODE REVIEW

File: Sources/Mesosphere/RValue.swift [hunk 1]
```diff
@@ -53,7 +53,11 @@ extension GIRGenFunction {
       argVals.reserveCapacity(args.count)
       for arg in args {
         let (newParent, value) = self.emitRValue(lastParent, arg)
-        argVals.append(value.copy(self).forward(self))
+        if value.value is Parameter {
```
Comment (CodaFi) [inline Sources/Mesosphere/RValue.swift:56]: I'm not 100% on this definition but it seems to work in practice. We don't necessarily want the basic blocks downstream of the chuzzling to destroy the values they extract from the `switch_constr` - the switch_constr's parent continuation still owns them transitively and they should be destroyed before an apply of the return continuation.

File: Sources/Mesosphere/Patterns.swift [hunk 2]
```diff
@@ -295,8 +295,25 @@ extension GIRGenFunction {
                       name: parent.name + "#col\(colIdx)row\(idx)")
         let destRef = self.B.createFunctionRef(destBB)
         //        assert(allIrrefutable(pats) != nil)
-        for _ in pats {
-          destBB.appendParameter(type: BottomType.shared)
+
+        guard let def = self.tc.signature.lookupDefinition(name.key) else {
+          fatalError()
```
Comment (harlanhaskins) [inline Sources/Mesosphere/Patterns.swift]: Even a single small string would help tremendously in these.

File: Sources/Mesosphere/RValue.swift (hunk 1 above)
Comment (harlanhaskins) [reply Sources/Mesosphere/RValue.swift:56]: 👍

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**